src/Apps/Order/Routes/Payment/OrderPaymentContext.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useReducer, useContext, createContext, Dispatch } from "react"
+import { CommercePaymentMethodEnum } from "__generated__/Payment_order.graphql"
+import { orderPaymentReducer } from "./orderPaymentReducer"
+import createLogger from "Utils/logger"
+
+const logger = createLogger("[dev: OrderPaymentContext] state:")
+
+/**
+ * Actions
+ */
+
+enum OrderPaymentActions {
+  SET_SELECTED_BANK_ACCOUNT_ID = "SET_SELECTED_BANK_ACCOUNT_ID",
+  SET_SELECTED_PAYMENT_METHOD = "SET_SELECTED_PAYMENT_METHOD",
+}
+
+type OrderPaymentActionsPayload = {
+  [OrderPaymentActions.SET_SELECTED_BANK_ACCOUNT_ID]: string
+  [OrderPaymentActions.SET_SELECTED_PAYMENT_METHOD]: CommercePaymentMethodEnum
+}
+
+type OrderPaymentAction = ActionMap<OrderPaymentActionsPayload>[keyof ActionMap<
+  OrderPaymentActionsPayload
+>]
+
+type ActionMap<M extends { [index: string]: any }> = {
+  [Key in keyof M]: M[Key] extends undefined
+    ? {
+        type: Key
+      }
+    : {
+        type: Key
+        payload: M[Key]
+      }
+}
+
+/**
+ * State
+ */
+
+type OrderPaymentState = {
+  selectedBankAccountId: string
+  selectedPaymentMethod: CommercePaymentMethodEnum | string
+}
+
+const initialOrderPaymentState = {
+  selectedBankAccountId: "",
+  selectedPaymentMethod: "",
+}
+
+/**
+ * Context
+ */
+
+const OrderPaymentContext = createContext<{
+  state: OrderPaymentState
+  dispatch: Dispatch<OrderPaymentAction>
+}>({
+  state: initialOrderPaymentState,
+  dispatch: () => null,
+})
+
+/**
+ * Provider
+ */
+
+const OrderPaymentProvider: React.FC = ({ children }) => {
```

Comment:
**nitpick:** Would `OrderPaymentContextProvider` be more accurate name for this component?

Reply:
def makes sense, I missed the consistency there. thank you!